src/project/mod.rs

```diff
@@ -620,6 +620,42 @@ impl Project {
         Ok(())
     }
 
+    /// Removes a dependency from `pixi.toml` based on `SpecType`.
+    pub fn remove_dependency(
```

Comment:
I think it is valuable to add some small tests to show that this does the correct modification as you showed on discord.
Something like this:
```rust
#[test]
fn test_remove_target_dependencies() {
  // Using known files in the project so the test succeed including the file check.
  let file_contents = r#"
      [dependencies]
      numpy = "*"

      [target.win-64.dependencies]
      numpy = "*"

      [target.linux-64.build-dependencies]
      numpy = "*"
      "#;

  let manifest = toml_edit::de::from_str::<ProjectManifest>(&format!(
      "{PROJECT_BOILERPLATE}\n{file_contents}"
  ))
      .unwrap();
  let mut project = Project::from_manifest(manifest);
  project.remove_target_dependency(&PackageName::try_from("numpy").unwrap(), &SpecType::Build ,&Platform::Linux64).unwrap();
  assert_debug_snapshot!(project.manifest);
}
```